File: php/features.php
<?php
declare(strict_types=1);

// CheckEMailOrUsernameForADataBreach SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CheckEMailOrUsernameForADataBreachFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CheckEMailOrUsernameForADataBreachBaseFeature();
            case "test":
                return new CheckEMailOrUsernameForADataBreachTestFeature();
            default:
                return new CheckEMailOrUsernameForADataBreachBaseFeature();
        }
    }
}

File: py/features.py
# CheckEMailOrUsernameForADataBreach SDK feature factory

from feature.base_feature import CheckEMailOrUsernameForADataBreachBaseFeature
from feature.test_feature import CheckEMailOrUsernameForADataBreachTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CheckEMailOrUsernameForADataBreachBaseFeature(),
        "test": lambda: CheckEMailOrUsernameForADataBreachTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
